src/sctools/metrics/metricsrunner.py

```diff
@@ -0,0 +1,61 @@
+import argparse
+import pysam
+
+
+class MetricsRunner:
+
+    def __init__(self, sam_file, open_mode=None):
+        """
+        :param str alignment_file: sam or bam file.
+        :param str open_mode: optional, mode to read file. Will be autodetected by file type if
+          the file contains the correct suffix for its type.
+        """
+
+        if open_mode is None:
+            if sam_file.endswith('.bam'):
+                open_mode = 'rb'
+            elif sam_file.endswith('.sam'):
+                open_mode = 'r'
+            else:
+                raise ValueError('could not autodetect file type for alignment_file %s '
+                                 '(detectible suffixes: .sam, .bam)' % sam_file)
+        self.input_sam = sam_file
+        self.open_mode = open_mode
+
+    def run_metrics(self, metrics_to_run):
+        """
+        :param list metrics_to_run: list of metrics to run for each record in input_sam
+        """
+
+        for metric in metrics_to_run:
+            metric.setup()
+
+        with pysam.AlignmentFile(self.input_sam, self.open_mode) as sam:
+            for record in sam:
+                for metric in metrics_to_run:
+                    metric.gather_metric(record)
+
+        for metric in metrics_to_run:
+            metric.calculate_and_output()
+
+
+def convert_class_name_to_class(metric_class_names):
+    # maybe some kind of case statement or python equivalent to grab a class from a string
+    # once we define a metric class that extends basemetric.py we can fill out this method
+    pass
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input_bam", help="path to input bam file")
+    parser.add_argument("-m", "--metric", action="append", help="metrics to run on input_bam")
```

Comment:
I'm a fan of `-m [list of whitespace-separated metrics]` vs `-m [m_1] -m [m_2] ... -m [m_n]`. What're your thoughts? 

In the former case, I'd make it `metrics` so that it's more obvious when referencing that we want multiple (see below)

Reply:
So the issue I had with`-m [list of whitespace-separated metrics]` is that it seemed you had to know how many inputs you were expecting but i took a second look at the docs and it seems its really easy.  I think I prefer being able to dump all the metrics at once for now so will change to `metrics`